internal/services/report_processor.go

```diff
@@ -75,122 +75,138 @@ func ProcessReport(msg *kafka.Message) {
+				return time_i.Before(time_j)
+			})
+
+			// looping over each group in chronological order based on interval_end time.
+			for _, group_name := range keys {
+				k8s_object := k8s_object_group_by_interval_end[group_name].Maps()
+				namespace := k8s_object[0]["namespace"].(string)
+				k8s_object_type := k8s_object[0]["k8s_object_type"].(string)
+				k8s_object_name := k8s_object[0]["k8s_object_name"].(string)
+				interval_start, err := utils.ConvertStringToTime(k8s_object[0]["interval_start"].(string))
+				if err != nil {
+					log.Errorf("unable to convert string to time: %s", err)
+					continue
+				}
+				interval_end, err := utils.ConvertStringToTime(k8s_object[0]["interval_end"].(string))
+				if err != nil {
+					log.Errorf("unable to convert string to time: %s", err)
+					continue
+				}
 
-			usage_data_byte, err := kruize.Update_results(experiment_name, k8s_object)
-			if err != nil {
-				log.Error(err)
-				continue
-			}
+				experiment_name := utils.GenerateExperimentName(
+					kafkaMsg.Metadata.Org_id,
+					kafkaMsg.Metadata.Source_id,
+					kafkaMsg.Metadata.Cluster_uuid,
+					namespace,
+					k8s_object_type,
+					k8s_object_name,
+				)
+
+				if workload_metrics, err := model.GetWorkloadMetricsForTimestamp(experiment_name, interval_end); err != nil {
+					log.Errorf("Error while checking for workload_metrics record: %s", err)
+					continue
+				} else if !reflect.ValueOf(workload_metrics).IsZero() {
+					log.Debugf("worload_metrics table already has data for interval_end time: %v.", interval_end)
+					continue
+				}
 
-			for _, container := range usage_data_byte[0].Kubernetes_objects[0].Containers {
-				container_usage_metrics, err := json.Marshal(container.Metrics)
+				container_names, err := kruize.Create_kruize_experiments(experiment_name, k8s_object)
 				if err != nil {
-					log.Errorf("Unable to marshal container usage data: %v", err)
+					log.Error(err)
+					continue
 				}
-				workload_metric := model.WorkloadMetrics{
-					WorkloadID:    workload.ID,
-					ContainerName: container.Container_name,
-					IntervalStart: interval_start,
-					IntervalEnd:   interval_end,
-					UsageMetrics:  container_usage_metrics,
+
+				// Create workload entry into the table.
+				workload := model.Workload{
+					ClusterID:       cluster.ID,
+					ExperimentName:  experiment_name,
+					Namespace:       namespace,
+					WorkloadType:    workload.WorkloadType(k8s_object_type),
+					WorkloadName:    k8s_object_name,
+					Containers:      container_names,
+					MetricsUploadAt: time.Now(),
 				}
-				if err := workload_metric.CreateWorkloadMetrics(); err != nil {
-					log.Errorf("unable to add record to workload_metrics table: %v. Error: %v", workload_metric, err)
+				if err := workload.CreateWorkload(); err != nil {
+					log.Errorf("unable to get or add record to workloads table: %v. Error: %v", workload, err)
```

Comment:
@patilsuraj767 Does workload.CreateWorkload have a native get or create functionality?

Reply:
It has create or update functionality, while inserting if it is found that workload is already present then it updates the `containers` and `metrics_upload_at` field of the same record. - https://github.com/RedHatInsights/ros-ocp-backend/blob/main/internal/model/workload.go#L36

Reply:
Okay, could you please modify the comment to `create or update`?

Reply:
```suggestion
					log.Errorf("unable to save workload record: %v. Error: %v", workload, err)
```
How about this?

Reply:
Done